anta/tests/bfd.py

```diff
@@ -0,0 +1,82 @@
+# Copyright (c) 2023-2024 Arista Networks, Inc.
+# Use of this source code is governed by the Apache License 2.0
+# that can be found in the LICENSE file.
+"""
+BFD test functions
+"""
+# Mypy does not understand AntaTest.Input typing
+# mypy: disable-error-code=attr-defined
+from __future__ import annotations
+
+from datetime import datetime
+from typing import Optional
+
+from anta.models import AntaCommand, AntaTest
+
+
+class VerifyBFDPeersHealth(AntaTest):
+    """
+    Verifies there is no IPv4 BFD peer in the down state, remote disc is not zero and last down should be above the threshold for all VRF.
+    Expected results:
+        * success: The test will pass if IPv4 BFD peers are not down, remote disc is not zero and last down above the defined threshold for all VRF.
+        * failure: The test will fail if IPv4 BFD peers are down, remote disc is zero and last down below the defined threshold for all VRF.
+    """
+
+    name = "VerifyBFDPeersHealth"
+    description = "Verifies there is no IPv4 BFD peer in the down state, remote disc is not zero and last down should be above the threshold for all VRF."
+    categories = ["bfd"]
+
+    # revision 1 as later revision introduces additional nesting for type
+    commands = [AntaCommand(command="show bfd peers", revision=1), AntaCommand(command="show clock")]
```

Comment:
Can you check if it would be possible to refactor the test with an updated revision?

Reply:
In revision 1 we will get bfd peer IP and in revision 2 we will get the source address also. So for parsing the bfd details we need to go through from source address. Left one belong to revision 1 and right one belong to revision 2
![Screenshot 2024-02-01 at 11 57 47](https://github.com/arista-netdevops-community/anta/assets/122076792/255e997f-97f4-4141-8197-d286a8bfa248)

Revision 3 and 4 showing the same structure but if compare to revision 1 or 2 then we get an additional root keys as type of session. Agin left one is revision 1 and right one is revision 3.
![Screenshot 2024-02-01 at 12 00 22](https://github.com/arista-netdevops-community/anta/assets/122076792/acf75163-e62a-46d2-8ba1-f07c133d70ae)

So i think revision 1 is the good one to parse the bfd details.